plugins/inputs/sqlserver/connectionstring.go

```diff
@@ -0,0 +1,124 @@
+package sqlserver
+
+import (
+	"net"
+	"net/url"
+	"strings"
+)
+
+const odbcPrefix = "odbc:"
+
+const (
+	emptyServerName   = "<empty-server-name>"
+	emptyDatabaseName = "<empty-database-name>"
+)
+
+// getConnectionIdentifiers returns the sqlInstance and databaseName from the given connection string.
+// The name of the server is returned as-is in the connection string
+// If the connection string could not be parsed or sqlInstance/databaseName were not present, a placeholder value is returned
+func getConnectionIdentifiers(connectionString string) (sqlInstance string, databaseName string) {
+	if len(connectionString) == 0 {
+		return
+	}
+
+	trimmedConnectionString := strings.TrimSpace(connectionString)
+
+	if strings.HasPrefix(trimmedConnectionString, odbcPrefix) {
+		i := strings.Index(trimmedConnectionString, odbcPrefix)
+		connectionStringWithoutOdbc := trimmedConnectionString[i+len(odbcPrefix):]
+		sqlInstance, databaseName = parseConnectionStringKeyValue(connectionStringWithoutOdbc)
+	} else if strings.HasPrefix(trimmedConnectionString, "sqlserver://") {
+		sqlInstance, databaseName = parseConnectionStringURL(trimmedConnectionString)
+	} else {
+		sqlInstance, databaseName = parseConnectionStringKeyValue(trimmedConnectionString)
+	}
+
+	return
```

Comment:
Please use
```suggestion
	if strings.HasPrefix(trimmedConnectionString, odbcPrefix) {
		i := strings.Index(trimmedConnectionString, odbcPrefix)
		connectionStringWithoutOdbc := trimmedConnectionString[i+len(odbcPrefix):]
		return parseConnectionStringKeyValue(connectionStringWithoutOdbc)
	}
	if strings.HasPrefix(trimmedConnectionString, "sqlserver://") {
		return parseConnectionStringURL(trimmedConnectionString)
	}
	return parseConnectionStringKeyValue(trimmedConnectionString)
```
to save some indention and make the code easier to follow.

Reply:
Good suggestion. Much cleaner.

Reply:
Please close as resolved.